### PR TITLE
Fetch configs from general source

### PR DESCRIPTION
**Pull request by shaunmaharaj**
Description:

- Initial push of refactor

Linting:

- [x] No linting errors

Tests:

- [x] Smoke tests (mvn clean install -Dcucumber.options="--tags @smoketest" from test)
- [x] Manual tests

Documentation:

- [x] Requires documentation updates

### CODE REVIEW

**Comment by shaunmaharaj** [comment]
@dusanradovanovic Wanted to get your opinion on this. I'll still be cleaning up the docker-compose file and the Jenkins jobs to match with this (where necessary). What do you think of using nginx' `sub_filter` feature to replace vars?

**Comment by shaunmaharaj** [comment]
@sminnath Can you please review the README? I made some updates to steps required for configuration updates.

File: README.md [hunk 1]
```diff
@@ -119,19 +119,19 @@ For more information about populating database, see the [Populating the Database
 
 ### Setting up a Production Environment
 1. Clone or pull the `react-pwa-reference-storefront` repository to your directory.
-2. Navigate to the `react-pwa-reference-storefront` directory.<br>
+2. Navigate to the `react-pwa-reference-storefront` directory.
+3. Add any custom configurations to the Dockerfile, Docker Compose file, and Nginx Configuration file
+    1. Provide any custom configurations you would like passed into your Docker container at runtime directly below the existing `RUN jq` command. One has bee provided to you as an example. Be sure to maintain the prefix of `NGINX_REPLACE_` for your custom configuration.
+    **Note:** You must use jq syntax for the command. More information on jq is availble [here](https://stedolan.github.io/jq/manual/).
+    2. Update the `./docker/prod/nginx.conf` file to include the new configuration directly below the existing one by adding a new `sub_filter` statement with your configuration within the configuration block. Be sure to expose this value with a more generic value, which will be used for the environment variable in the next step.
```
Comment (sminnath) [inline README.md]:
```suggestion
    2. Update the `./docker/prod/nginx.conf` file with the new configuration by adding a new `sub_filter` statement within the configuration block below the existing configuration. 
    
    Use a generic value, which is used for the environment variable in the next step, for the configuration.
```

File: README.md [hunk 2]
```diff
@@ -119,19 +119,19 @@ For more information about populating database, see the [Populating the Database
...
+    3. Update the `./docker/prod/docker-compose.yaml` file to include your environment configuration variable from step 2 into the `environment` block, and set it to the value you desire to pass into your container.
```
Comment (sminnath) [inline README.md]:
```suggestion
    3. In the `./docker/prod/docker-compose.yaml` file, add the environment configuration variable within the `environment` block and set a value to add in the container.
```

File: README.md [hunk 3]
```diff
@@ -119,19 +119,19 @@ For more information about populating database, see the [Populating the Database
...
 3. Build a Production Docker Image:
```
Comment (sminnath) [inline README.md]:
```suggestion
3. Build a Production Docker Image by performing the following:
```

File: README.md [hunk 4]
```diff
@@ -119,19 +119,19 @@ For more information about populating database, see the [Populating the Database
...
-    1. Clone or pull the `react-pwa-reference-storefront` repository to your directory.
-    2. Run the `cd react-pwa-reference-storefront` command.
-    3. Run the `docker build -t ep-store -f ./docker/prod/Dockerfile .` command.
-    4. Push the `ep-store` image to the docker repository.
-
+    1. Run the `docker build -t ep-store -f ./docker/prod/Dockerfile .` command.
+    2. Push the `ep-store` image to the docker repository.
```
Comment (sminnath) [inline README.md]:
```suggestion
    2. Push the `ep-store` image to the Docker repository.
```

File: README.md [hunk 5]
```diff
@@ -119,19 +119,19 @@ For more information about populating database, see the [Populating the Database
...
 4. Run a Production Docker Image:
```
Comment (sminnath) [inline README.md]:
```suggestion
4. Run a Production Docker Image, by performing the following :
```

File: README.md [hunk 6]
```diff
@@ -119,19 +119,19 @@ For more information about populating database, see the [Populating the Database
...
     1. Navigate to the `docker/prod/` directory.<br>
-    2. Copy the following files to the user home directory on the remote host:<br>
-        * `docker-compose.yaml`
-        * `nginx.conf`
-    3. In the `nginx.conf` file, update the following parameters:
+    2. In the `docker-compose.yaml` file, update the following parameters:
+        * `$DOCKER_REPO` with `ep-store`.
```
Comment (sminnath) [inline README.md]: Use a table here

Parameter                             Value 
`$DOCKER_REPO`              `ep-store`
`$CORTEX_URL`                  The Cortex server URL.
`$AUTHSERVICE_URL`        The Cortex Account Management service URL. (This  parameter is required for all scenarios. For a B2C storefront, set this value to the`$CORTEX_URL` parameter).

`$DOMAIN`                             The domain name without `http://`.       


use table for all the parameters

File: README.md [hunk 7]
```diff
@@ -140,13 +140,8 @@ For more information about populating database, see the [Populating the Database
...
         **Note:** You can use the certificate generated by any certificate provider, such as [Let's Encrypt](https://letsencrypt.org).
-    4. In the `docker-compose.yaml` file, update the following parameters:
-        * `$DOCKER_REPO` with `ep-store`.
-        * `$SSL_CERT_PATH` with the certificate file path in the remote server .
-        <br> For example, `/etc/letsencrypt/live/reference.elasticpath.com/fullchain.pem`.<br>
-        * `$SSL_KEY_PATH` with the path of the private key in the remote server.
-        <br> For example, `/etc/letsencrypt/live/reference.elasticpath.com/privkey.pem`.<br>
-    5. Run the following Docker command to start the Docker container created for the Docker image:<br/>
+        * `$CORTEX_API_SCOPE` with the name of the store from which Cortex retrieves data.
+    3. Run the following Docker command to start the Docker container created for the Docker image:<br/>
```
Comment (sminnath) [inline README.md]:
```suggestion
    3. To start the Docker container created for the Docker image, run the following Docker command:
```

File: docker/prod/Dockerfile [hunk 8]
```diff
@@ -6,7 +6,23 @@ ADD ./package.json /project
 ADD ./package-lock.json /project
 RUN npm install
 ADD ./ /project
+
+# Install jq for inline JSON replacement
```
Comment (sminnath) [inline docker/prod/Dockerfile]:
```suggestion
# Installing jq for inline JSON replacement
```

File: docker/prod/Dockerfile [hunk 9]
```diff
@@ -6,7 +6,23 @@ ADD ./package.json /project
...
+RUN apt-get update
```
Comment (sminnath) [inline docker/prod/Dockerfile]: Run the following commands: 

1.  To ...., command.
write in this format

**Pull request merged**